File: app/src/main/java/com/zaclimon/tsutaeru/ui/components/cardview/PicassoCardViewImageProcessor.java
package com.zaclimon.tsutaeru.ui.components.cardview;

import android.widget.ImageView;

import com.squareup.picasso.Picasso;
import com.zaclimon.xipl.ui.components.cardview.CardViewImageProcessor;

/**
 * Concrete implementation of {@link CardViewImageProcessor} that uses Picasso as
 * an underlying backend library.
 *
 * @author zaclimon
 * Creation date: 07/07/17
 */

public class PicassoCardViewImageProcessor implements CardViewImageProcessor {

    @Override
    public void loadImage(String url, int cardViewWidth, int cardViewHeight, ImageView imageView) {
        Picasso.with(imageView.getContext()).load(url).resize(cardViewWidth, cardViewHeight).into(imageView);
    }

}

File: app/src/main/java/com/zaclimon/tsutaeru/ui/components/cardview/PicassoCardViewImageProcessor.kt
package com.zaclimon.tsutaeru.ui.components.cardview

import android.widget.ImageView
import com.squareup.picasso.Picasso
import com.zaclimon.xipl.ui.components.cardview.CardViewImageProcessor

/**
 * Concrete implementation of [CardViewImageProcessor] that uses Picasso as
 * an underlying backend library.
 *
 * @author zaclimon
 */
class PicassoCardViewImageProcessor : CardViewImageProcessor {

    override fun loadImage(url: String?, cardViewWidth: Int, cardViewHeight: Int, imageView: ImageView?) {
        Picasso.with(imageView?.context).load(url).resize(cardViewWidth, cardViewHeight).into(imageView)
    }

}